interface/src/app/node/node.js

```diff
@@ -304,6 +307,32 @@ angular.module('adage.node', [
         $scope.numExpShown =
           $scope.topMode ? $scope.topNum : $scope.experiments.length;
       };
+
+      // Event handler when user clicks the checkbox in front of each sample.
+      // It keeps track of the number of selected samples in each experiment
+      // to indicate whether the "show annotations" button should be enabled.
+      $scope.toggleSelection = function(exp, selected) {
+        if (selected) {
+          ++exp.numSelections;
+        } else {
+          --exp.numSelections;
+        }
+      };
```

Comment:
Right. The thing to keep in mind is that memory is also a constraint (along with CPU) in the browser context. Although modern hardware is fairly capacious there, too, browsers need to juggle dozens of tabs and keep from getting bogged down by the JavaScript on *all* of those pages operating simultaneously. If the page needs to run on a mobile browser (not really a consideration for us yet), everything needs to be that much tighter.